api/src/integrations/core/datasource.js

```diff
@@ -295,57 +309,132 @@ class CoreDataSource {
-            }
+    return validPermit;
+  }
 
-            validPermit = permit;
-          }
-        }
-      }
+  /**
+   * Creates a new Mine Permit record.
+   * 
+   * @param {object} permitUtils Mine Permit utilities.
+   * @param {*} nrptiRecord Transformed Core record.
+   * @returns {MinePermit} Newly created Mine Permit.
+   * @memberof CoreDataSource
+   */
+  async createMinePermit(permitUtils, permitAmendmentUtils, nrptiRecord) {
+    if (!permitUtils) {
+      throw new Error('createMinePermit - permitUtils is required');
+    }
 
-      return validPermit;
-    } catch(error) {
-      defaultLog.error(`getMinePermit - unexpected error: ${error.message}`);
-      throw(error);
+    if (!permitAmendmentUtils) {
+      throw new Error('createMinePermit - permitAmendmentUtils is required');
+    }
+
+    if (!nrptiRecord) {
+      throw new Error('createMinePermit - nrptiRecord is required');
+    }
+
+    const permit = await this.getMinePermit(nrptiRecord);
+
+    if (!permit) {
+      throw new Error('createMinePermit - Cannot find valid permit');
+    }
+
+    // Create the permit amendments that belong to the permit.
+    const transformedAmendments = await permitAmendmentUtils.transformRecords(permit);
+    // To trigger flavour for this import.
+    const preparedAmendments = transformedAmendments.map(amendment => ({ ...amendment, PermitAmendmentBCMI: {} }))
+    const savedAmendments = await permitAmendmentUtils.createRecord(preparedAmendments);
+ 
+    // Transform and create the permit record.
+    const transformedPermit = await permitUtils.transformRecord(permit, savedAmendments);
+    // To trigger flavour for this import.
+    transformedPermit.PermitBCMI = {};
+    const savedRecord = await permitUtils.createRecord(transformedPermit);
+
+    // Get the newly created permit.
+    return savedRecord[0] &&
+            savedRecord[0].object &&
+            savedRecord[0].object[0] &&
+            savedRecord[0].object[0]
+  }
+
+  /**
+   * Updates the Mine Permit record.
+   * 
+   * @param {object} permitUtils Mine Permit utilities
+   * @param {Mine} mineRecord Existing Mine record
+   * @returns {boolean} Indication of success or not
+   * @memberof CoreDataSource
+   */
+  async updateMinePermit(permitUtils, permitAmendmentUtils, mineRecord) {
+    // Get the updated Core permit.
+    const permit = await this.getMinePermit(mineRecord);
+
+    if (!permit) {
+      throw new Error('updateMinePermit - Cannot find valid permit');
     }
+
+    // Update the amendments and permit.
+    const transformedAmendments = await permitAmendmentUtils.transformRecords(permit);
+    // To trigger flavour for this import.
+    const preparedAmendments = transformedAmendments.map(amendment => ({ ...amendment, PermitAmendmentBCMI: {} }))
+    const updatedAmendments = await permitAmendmentUtils.updateRecord(preparedAmendments);
```

Comment:
Is it possible to be performing an update to an existing Permit, but have new amendments that need to get created?
If so, we probably need to do the findExisting check here first, and the call either permitAmendmentUtils update/create.

Reply:
Good call. Updated to handle an update and create.